dev-tools/mage/pkgtypes.go

```diff
@@ -711,7 +711,14 @@ func addUidGidEnvArgs(args []string) ([]string, error) {
 
 // addFileToZip adds a file (or directory) to a zip archive.
 func addFileToZip(ar *zip.Writer, baseDir string, pkgFile PackageFile) error {
-	return filepath.Walk(pkgFile.Source, func(path string, info os.FileInfo, err error) error {
+	// filepath.Walk() does not resolve symlinks, but pkgFile.Source might be one,
+	// see mage.KibanaDashboards().
+	resolvedSource, err := filepath.EvalSymlinks(pkgFile.Source)
```

Comment:
> This will prevent the usage of symlinks in packages.

Sorry, I don't follow. This was meant to allow `PackageFile.Source` to be a symlink to a directory. This is never the case at the moment - when I changed it to create a symlink from `_meta/kibana.generated` to `build/kibana` the build failed and I had to make the change here to allow it. So I'm not sure how it prevents symlinks?

The problem is that `make snapshot` depends on `make beats-dashboards` (via `make release`) which is implemented as `mage packageBeatDashboards` which assumes all Beats have a `_meta/kibana.generated` directory:

https://github.com/elastic/beats/blob/feature-auditbeat-host/magefile.go#L64
```
for _, beat := range Beats {
		spec.Files[beat] = mage.PackageFile{
			Source: filepath.Join(beat, "_meta/kibana.generated"),
		}
	}

	return mage.PackageZip(spec.Evaluate())
```

This would have to be changed to take Beats specifics into account. E.g. Auditbeat produces a `build/kibana` directory, while Filebeat still has `_meta/kibana.generated`. I believe `PackageKibanaDashboardsFromBuildDir()` just changes everything, regardless of what each Beat needs. So I'm not sure if we can currently do this without a bigger effort? Symlinks seemed like the simpler way.

Reply:
Sorry, I didn’t realize this was solving for the beats dashboards zip case. (one the challenges of reviewing from a phone)

No one is adding symlinks via packages afaik (e.g. like installing a symlink on the system that is owned by an rpm/deb package) and I’m not sure that works anyways with our cade. So don’t consider this a problem. And once all Beats are writing dashboards to build/kibana this won’t be necessary anyways.

However this does make me wonder if dashboards from x-pack/auditbeat (or x-pack/filebeat) are being bundled in the beat dashboard zip. I suspect not, but that’s also not a problem for this PR to solve.